tests/unit/IonTextReaderTest.js

```diff
@@ -1,130 +1,171 @@
+
+            ionReader.next();
+
+            assert.equal(ionReader.fieldName(), "b");
+
+            ionReader.stepIn(); // Step into the "b" struct.
+            ionReader.next();
+
+            assert.equal(ionReader.fieldName(), "key2");
+            assert.equal(ionReader.value(), "string2");
+        };
+
+        suite['Parse through struct can skip over nested containers'] = function() {
+            var ionToRead = "{ outerkey1 : { innerkey1 : {a1: \"a1\", b1: \"b1\"} }, outerkey2 : { innerkey2 : {a2: \"a2\", b2: \"b2\"} } }";
+            var ionReader = ion.makeReader(ionToRead);
+            ionReader.next();
+
+            assert.equal(ion.IonTypes.STRUCT, ionReader.valueType());
+
+            ionReader.stepIn(); // Step into the base struct.
+            ionReader.next();
+
+            assert.equal(ionReader.fieldName(), "outerkey1");
+
+            ionReader.next();
+
+            assert.equal(ionReader.fieldName(), "outerkey2");
+
+            ionReader.stepIn(); // Step into the "b" struct.
+            ionReader.next();
+
+            assert.equal(ionReader.fieldName(), "innerkey2");
+        };
+
+
+        suite['text IVM'] = function() {
+            var textReader = ion.makeReader("");
+            let isNotIVM = ["$ion_schema_1_0", "$ion_1", "$ion_1_a", "$ion_", "ion_1_"];
+            let unsupportedIVM = ["$ion_2_0", "$ion_1_999", "$ion_999_0", "$ion_1_1", "$ion_1_00"];
+
+            test_isIVM(textReader,"$ion_1_0", true, 0, []);
+            test_isIVM(textReader,"$ion_1_0", false, 1, []);
+            test_isIVM(textReader,"$ion_1_0", false, 0, ["taco"]);
+            for (let i = 0; i < isNotIVM.length; i++) {
+                test_isIVM(textReader, isNotIVM[i], false, 0, []);
+            }
+            for (let i = 0; i < unsupportedIVM.length; i++) {
+                test_isIVM(textReader, unsupportedIVM[i], "throw", 0, []);
+            }
+            for (let i = 0; i < unsupportedIVM.length; i++) {
+                test_isIVM(textReader, unsupportedIVM[i], false, 1, []);
+            }
+            for (let i = 0; i < unsupportedIVM.length; i++) {
+                test_isIVM(textReader, unsupportedIVM[i], false, 0, ["taco"]);
+            }
+        };
+
+        function test_isIVM(reader, value, expected, depth, annotations) {
+            if (expected === "throw") {
+                try {
+                    reader.isIVM(value, depth, annotations);
+                } catch(error) {
+                    if (error.message !== "Only Ion version 1.0 is supported.") {
+                        throw new Error(`input: ${value} threw an unexpected error: ${error.message}.`);
+                    }
+                    return;
+                }
+                throw new Error("Expected " + value + " to throw an error.");
+            } else {
+                let actual = reader.isIVM(value, depth, annotations);
+                assert.equal(expected, actual);
```

Comment:
chai's [assert.equal](https://www.chaijs.com/api/assert/#method_equal) signature is `assert.equal(actual, expected)` -- where `actual` and `expected` are in alphabetical order (this is the opposite of JUnit's ordering, fyi).  please follow chai's signature wrt parameter ordering--otherwise, test failure messages are likely to cause confusion

Reply:
please change this line to:  `assert.equal(actual, expected)`